app/main.py

```diff
@@ -1,3 +1,66 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: [int, "Distance"]) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        else:
+            return Distance(self.km + other)
+
+    def __iadd__(self, other: [int, "Distance"]) -> "Distance":
```

Comment:
The type hint for the `other` parameter should be `Union[int, Distance]` instead of `[int, "Distance"]` to properly indicate that `other` can be either an `int` or a `Distance` instance.